client/state/analytics/middleware.js

```diff
@@ -0,0 +1,40 @@
+import analytics from 'analytics';
+import invoke from 'lodash/invoke';
+import reduxAnalytics from 'redux-analytics';
```

Comment:
I started from scratch initially and felt the same way, but because it was already being handled in Calypso with `redux-analytics`, and since `redux-analytics` _does_ provide a decent API, I wanted to explore doing it this way.

After coding it up, I'm not too disappointed with this method. It hides the rest of the action from what we care about and focusses more on the metadata.